token-metadata/program/tests/utils/digital_asset.rs

```diff
@@ -64,26 +65,146 @@ impl DigitalAsset {
         }
     }
 
+    // Note the authority is the payer of the transaction.
+    pub async fn verify(
+        &mut self,
+        context: &mut ProgramTestContext,
+        authority: Keypair,
+        args: VerifyArgs,
+        metadata: Option<Pubkey>,
+        delegate_record: Option<Pubkey>,
+        collection_mint: Option<Pubkey>,
+        collection_metadata: Option<Pubkey>,
+        collection_master_edition: Option<Pubkey>,
+    ) -> Result<(), BanksClientError> {
+        let mut builder = VerifyBuilder::new();
+        builder
+            .authority(authority.pubkey())
+            .metadata(metadata.unwrap_or(self.metadata));
+
+        match args {
+            VerifyArgs::CreatorV1 => (),
+            VerifyArgs::CollectionV1 => {
+                if let Some(delegate_record) = delegate_record {
+                    builder.delegate_record(delegate_record);
+                }
+
+                if let Some(collection_mint) = collection_mint {
+                    builder.collection_mint(collection_mint);
+                }
+
+                if let Some(collection_metadata) = collection_metadata {
+                    builder.collection_metadata(collection_metadata);
+                }
+
+                if let Some(collection_master_edition) = collection_master_edition {
+                    builder.collection_master_edition(collection_master_edition);
+                }
+            }
+        }
+
+        let verify_ix = builder.build(args).unwrap().instruction();
+
+        let transaction = Transaction::new_signed_with_payer(
+            &[verify_ix],
+            Some(&authority.pubkey()),
+            &[&authority],
+            context.last_blockhash,
+        );
+
+        context.banks_client.process_transaction(transaction).await
+    }
+
+    // Note the authority is the payer of the transaction.
+    pub async fn unverify(
+        &mut self,
+        context: &mut ProgramTestContext,
+        authority: Keypair,
+        args: VerifyArgs,
+        metadata: Option<Pubkey>,
+        delegate_record: Option<Pubkey>,
+        collection_mint: Option<Pubkey>,
+        collection_metadata: Option<Pubkey>,
+    ) -> Result<(), BanksClientError> {
+        let mut builder = UnverifyBuilder::new();
+        builder
+            .authority(authority.pubkey())
+            .metadata(metadata.unwrap_or(self.metadata));
+
+        match args {
+            VerifyArgs::CreatorV1 => (),
```

Comment:
After https://github.com/metaplex-foundation/metaplex-program-library/commit/ae882ee7fe85e8e3ba566f93f2968df7ea39eee9 we don't let them remove a verified creator either by setting creators array to `None` or setting creators array to `Some(Vec<Creator>)`.  So they need to unverify first before removal.

I'm not 100% sure what you mean by this comment though, because this unverify function is just building the instruction, and in the case of unverifying creators, it is specified using `VerifyArgs::CreatorV1` and then there's nothing else in the instruction to add (no delegate record, collection mint, collection metadata needed when unverifying a creator).

One thing I'm doing in the new version is renaming `VerifyArgs` to `VerificationArgs` to make it clear that the same args struct is used by both `Verify` and `Unverify`.

Reply:
Looks good.